### PR TITLE
[ADD] Security Test Setup

### PR DESCRIPTION
**Pull request by eilst**

### CODE REVIEW

File: project_event/tests/test_security.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                if m.name == menu_name:
+                    return True
+        return False
+
+    def get_user_groups(self, user_id):
+        user = self.env['res.users'].browse(user_id)
+        return user.groups_id
+
+    def get_group_access_rules(self, group_id):
+        group = self.env['res.groups'].brwose(group_id)
```
Comment (rimbendhaou) [inline project_event/tests/test_security.py]: typo browse

File: project_resource_calendar/tests/test_security.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                if m.name == menu_name:
+                    return True
+        return False
+
+    def get_user_groups(self, user_id):
+        user = self.env['res.users'].browse(user_id)
+        return user.groups_id
+
+    def get_group_access_rules(self, group_id):
+        group = self.env['res.groups'].brwose(group_id)
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_security.py]: typo browse

File: project_resource_calendar/tests/test_security.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        return group.model_access
+    
+    def has_read_permission(self, access_rule):
+        return access_rule.perm_read        
+      
+    def cannot_create_room(self, user_id, room_name):
+        with self.assertRaises(exceptions.AccessError):
+            self.env['resource.calendar.room'].sudo(user_id).create({'name': room_name})
+
+    def test_010_base_user_can_not_crete_rooms(self):
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_security.py]: typo crete -->create

File: project_resource_calendar/tests/test_security.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        self.cannot_create_room(self.user_base.id,'Room base')
+    
+    def test_011_guest_user_can_not_crete_rooms(self):
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_security.py]: typo crete -->create

File: project_resource_calendar/tests/test_security.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        self.cannot_create_room(self.user_guest.id,'Room guest')
+    
+    def test_012_editor_user_can_not_crete_rooms(self):
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_security.py]: typo crete -->create

File: project_resource_calendar/tests/test_security.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        self.cannot_create_room(self.user_editor.id,'Room editor')
+    
+    def test_013_manager_user_can_create_rooms(self):
+        room_manager = self.env['resource.calendar.room'].sudo(
+            self.user_manager).create({'name': 'Manager Room'})
+        self.assertIsInstance(room_manager,type(self.env['resource.calendar.room']))
+        
+    def test_020_base_user_can_not_crete_instruments(self):
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_security.py]: typo crete -->create

File: project_resource_calendar/tests/test_security.py [hunk 7]
```diff
@@ -0,0 +1,116 @@
+# © 2018 Savoir-faire Linux
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_security.py]: 2019

File: project_event/tests/test_security.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
+# © 2018 Savoir-faire Linux
```
Comment (rimbendhaou) [inline project_event/tests/test_security.py]: 2019

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_resource_calendar/tests/test_security.py [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    
+    def test_012_editor_user_can_not_create_rooms(self):
+        self.cannot_create_room(self.user_editor.id,'Room editor')
+    
+    def test_013_manager_user_can_create_rooms(self):
+        room_manager = self.env['resource.calendar.room'].sudo(
+            self.user_manager).create({'name': 'Manager Room'})
+        self.assertIsInstance(room_manager,type(self.env['resource.calendar.room']))
+    
+    def test_20_base_user_cannot_read_calendar_events_where_he_is_not_participant(self):
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_security.py]: test_20_base_user_cannot_read_calendar_events_where_he_is_not_participant failed: AssertionError: AccessError not raised

File: project_event/tests/test_security.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+                'user_id': self.user_editor.id
+        })
+        self.employee_manager = self.Employees.create({
+                'name': 'Manager',
+                'work_email': 'manager@test.com',
+                'user_id': self.user_manager.id
+        })
+
+
+    def has_access_to_menu(self, user_id, menu_name):
```
Comment (rimbendhaou) [inline project_event/tests/test_security.py:50]: PEP8 too many blank lines

File: project_event/tests/test_security.py [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+        self.PROJECT_EVENT_EDITOR = self.env.ref(
+            'project_event.group_project_event_editor')
+        self.PROJECT_EVENT_MANAGER = self.env.ref(
+            'project_event.group_project_event_manager')
+        self.Users = self.env['res.users']
+        self.Employees = self.env['hr.employee']
+        self.project_user = self.Users.create({
+                'name': 'User',
+                'login': 'base@test.com',
+                'groups_id': [(6, 0,[self.PROJECT_EVENT_USER.id] )]
```
Comment (rimbendhaou) [inline project_event/tests/test_security.py]: PEP8 space after comma
no space after [self.PROJECT_EVENT_USER.id]

File: project_event/tests/test_security.py [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+            })
+        self.user_editor = self.Users.create({
+                'name': 'Editor',
+                'login': 'editor@test.com',
+                'groups_id': [(6, 0,[self.PROJECT_EVENT_EDITOR.id] )]
```
Comment (rimbendhaou) [inline project_event/tests/test_security.py]: PEP8 space after comma
no space after [self.PROJECT_EVENT_USER.id]

File: project_event/tests/test_security.py [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+            })
+        self.user_manager = self.Users.create({
+                'name': 'Manager',
+                'login': 'manager@test.com',
+                'groups_id': [(6, 0,[self.PROJECT_EVENT_MANAGER.id] )]
```
Comment (rimbendhaou) [inline project_event/tests/test_security.py]: PEP8 space after comma
no space after [self.PROJECT_EVENT_USER.id]

File: project_event/tests/test_security.py [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+    def has_access_to_menu(self, user_id, menu_name):
+        user = self.env['res.users'].browse(user_id)
+        for group in user.groups_id:
+            for m in group.menu_access:
+                if m.name == menu_name:
+                    return True
+        return False
+
+    def get_user_groups(self, user_id):
```
Comment (rimbendhaou) [inline project_event/tests/test_security.py:58]: these methods will be useful in many unit tests of different modules, better to add these method in the model (user or group). Check first if we dont have in odoo a similar method

File: project_resource_calendar/tests/test_security.py [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+            'project_resource_calendar.group_resource_calendar_manager')
+        self.RESOURCE_CALENDAR_EDITOR = self.env.ref(
+            'project_resource_calendar.group_resource_calendar_editor')
+        self.Users = self.env['res.users']
+        self.Tag_categories = self.env['hr.employee.category']
+        self.Employees = self.env['hr.employee']
+        self.user_base = self.Users.create({
+                'name': 'Base User',
+                'login': 'base@test.com',
+                'groups_id': [(6, 0,[self.EMPLOYEE_BASE.id] )]
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_security.py]: PEP8 space after comma
no space after .id]

Comment (rimbendhaou) [reply project_resource_calendar/tests/test_security.py]: Same comment in the above code

File: project_resource_calendar/tests/test_security.py [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+    def test_011_guest_user_can_not_create_rooms(self):
+        self.cannot_create_room(self.user_guest.id,'Room guest')
+    
+    def test_012_editor_user_can_not_create_rooms(self):
+        self.cannot_create_room(self.user_editor.id,'Room editor')
+    
+    def test_013_manager_user_can_create_rooms(self):
+        room_manager = self.env['resource.calendar.room'].sudo(
+            self.user_manager).create({'name': 'Manager Room'})
+        self.assertIsInstance(room_manager,type(self.env['resource.calendar.room']))
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_security.py]: PEP8 space after comma

File: project_resource_calendar/tests/test_security.py [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+            self.env['resource.calendar.room'].sudo(user_id).create({'name': room_name})
+
+    def test_010_base_user_can_not_create_rooms(self):
+        self.cannot_create_room(self.user_base.id,'Room base')
+    
+    def test_011_guest_user_can_not_create_rooms(self):
+        self.cannot_create_room(self.user_guest.id,'Room guest')
+    
+    def test_012_editor_user_can_not_create_rooms(self):
+        self.cannot_create_room(self.user_editor.id,'Room editor')
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_security.py]: PEP8 space after comma

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_resource_calendar/tests/test_security.py (hunk 2 above)
Comment (eilst) [reply project_resource_calendar/tests/test_security.py]: done

File: project_event/tests/test_security.py (hunk 1 above)
Comment (eilst) [reply project_event/tests/test_security.py]: done

File: project_resource_calendar/tests/test_security.py (hunk 3 above)
Comment (eilst) [reply project_resource_calendar/tests/test_security.py]: done

File: project_resource_calendar/tests/test_security.py (hunk 4 above)
Comment (eilst) [reply project_resource_calendar/tests/test_security.py]: done

File: project_resource_calendar/tests/test_security.py (hunk 5 above)
Comment (eilst) [reply project_resource_calendar/tests/test_security.py]: done

File: project_resource_calendar/tests/test_security.py (hunk 6 above)
Comment (eilst) [reply project_resource_calendar/tests/test_security.py]: done

File: project_resource_calendar/tests/test_security.py (hunk 7 above)
Comment (eilst) [reply project_resource_calendar/tests/test_security.py]: yes, done

File: project_event/tests/test_security.py (hunk 8 above)
Comment (eilst) [reply project_event/tests/test_security.py]: done

File: project_resource_calendar/tests/test_security.py (hunk 9 above)
Comment (eilst) [reply project_resource_calendar/tests/test_security.py]: done. was the wrong assert

File: project_event/tests/test_security.py (hunk 11 above)
Comment (eilst) [reply project_event/tests/test_security.py]: done

File: project_event/tests/test_security.py (hunk 12 above)
Comment (eilst) [reply project_event/tests/test_security.py]: done

File: project_event/tests/test_security.py (hunk 13 above)
Comment (eilst) [reply project_event/tests/test_security.py]: done

Comment (eilst) [reply project_event/tests/test_security.py]: already added pep8 check before allowing me commit something :+1:

**Review by rimbendhaou — APPROVED** [review]

**Pull request merged**